### PR TITLE
CI: Add hs4x-pae40 to CI testing

### PR DESCRIPTION
**Pull request by VVIsaev**
Signed-off-by: Vladimir Isaev <isaev@synopsys.com>

### CODE REVIEW

**Review by desmodrome — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,7 +72,16 @@ jobs:
           export CROSS_COMPILE=${{ github.workspace }}/toolchain/bin/${{ matrix.targets.toolchain }}-linux-gnu-
 
           make ${{ matrix.targets.defconfig }}
-          ./scripts/config --set-str CONFIG_INITRAMFS_SOURCE ${{ github.workspace }}/arc-kernel-ci/cpio/rootfs.cpio.${{ matrix.targets.cpu }}
+
+          ROOTFS_PATH=${{ github.workspace }}/arc-kernel-ci/cpio/rootfs.cpio.${{ matrix.targets.cpu }}
+          if [[ "${{ matrix.targets.cpu }}" == "hs4x-pae40" ]]; then
+            ROOTFS_PATH=${{ github.workspace }}/arc-kernel-ci/cpio/rootfs.cpio.hs4x
```
Comment (abrodkin) [inline .github/workflows/ci.yml]: Maybe do explicit selection of `rootfs.xxx` for each CPU type?
That way we may add more CPU flavors later w/o such overrides.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -69,7 +72,16 @@ jobs:
...
+          fi
+          ./scripts/config --set-str CONFIG_INITRAMFS_SOURCE "$ROOTFS_PATH"
+
+          if [[ "${{ matrix.targets.cpu }}" == "hs4x-pae40" ]]; then
+            ./scripts/config --enable CONFIG_ARC_HAS_PAE40
```
Comment (abrodkin) [inline .github/workflows/ci.yml:95]: Maybe even add this into the same "switch/case" as the change above i.e.:
```python
ROOTFS_PATH=${{ github.workspace }}/arc-kernel-ci/cpio/rootfs.cpio.hs4x
./scripts/config --enable CONFIG_ARC_HAS_PAE40

...

./scripts/config --set-str CONFIG_INITRAMFS_SOURCE "$ROOTFS_PATH"

```

**Review by abrodkin — COMMENTED** [review]
See my comments in code

**Review by geomatsi — APPROVED** [review]
Reviewed-by: Sergey Matyukevich <sergey.matyukevich@synopsys.com>

File: .github/workflows/ci.yml (hunk 2 above)
Comment (VVIsaev) [reply .github/workflows/ci.yml:95]: I want to separate rootfs selection and other options selection. So rootfs selection can be moved to script/function (get_rootf_path()) in future if we will have much more CPU types.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (VVIsaev) [reply .github/workflows/ci.yml]: done!

**Review by abrodkin — APPROVED** [review]

**Pull request merged**